src/python/pygrackle/grackle_wrapper.pyx

```diff
@@ -872,8 +878,7 @@ cdef class chemistry_data:
 
     property temperature_units:
         def __get__(self):
-            return mass_hydrogen_cgs * \
-              self.velocity_units**2 / boltzmann_constant_cgs
+            return get_temperature_units(&self.units)
```

Comment:
This change is entirely responsible for the test failures. This causes a relative change in the temperature units of around 7e-4 from the different values of mh and kb used internally and from yt's physical constants. I will update the test answers and merge.

Reply:
Yes, that's makes sense -- I think this is good to go!